engine/src/main/java/org/terasology/network/BroadCastServer.java

```diff
@@ -45,26 +44,29 @@ public void startBroadcast() {
             Executors.newSingleThreadExecutor().execute(new Runnable() {
                 @Override
                 public void run() {
-                    while (true) {
+                    while (!Thread.currentThread().isInterrupted()) {
                         try {
                             while (turnOnBroadcast) {
                                 listenToBroadCast();
                             }
                         } catch (Exception e) {
                             logger.error("Broadcasting has been interrupted" + e.getMessage());
+                            stopBroadCast();
                         }
                     }
                 }
             });
         } catch (Exception e) {
         logger.info("Broadcasting has been interrupted " + e.getMessage());
+        stopBroadCast();
```

Comment:
@DarkWeird Stopping broadcast on interruption would close the sockets thereby preventing unnecessary open sockets